### PR TITLE
ARROW-15760: [C++] Avoid hard dependency on git in cmake (download tarballs from github instead)

### PR DESCRIPTION
**Pull request by mbrobbel**
As suggested in https://github.com/apache/arrow/pull/12322#issuecomment-1048527114.

### CODE REVIEW

File: cpp/src/arrow/engine/CMakeLists.txt [hunk 1]
```diff
@@ -48,8 +48,7 @@ set(SUBSTRAIT_PROTOS
     type_expressions)
 
 externalproject_add(substrait_ep
-                    GIT_REPOSITORY "${ARROW_SUBSTRAIT_REPO}"
-                    GIT_TAG "${ARROW_SUBSTRAIT_TAG}"
+                    URL "https://github.com/${ARROW_SUBSTRAIT_REPO}/archive/${ARROW_SUBSTRAIT_TAG}.tar.gz"
```
Comment (jorisvandenbossche) [inline cpp/src/arrow/engine/CMakeLists.txt:51]:
```suggestion
                    URL "${ARROW_SUBSTRAIT_REPO}/archive/${ARROW_SUBSTRAIT_TAG}.tar.gz"
```

(the https://github.com/ is included in the variable)

Comment (kszucs) [reply cpp/src/arrow/engine/CMakeLists.txt]: This url will only work for github so we my want to restore the previous url and note that `ARROW_SUBSTRAIT_REPO` should be a github `user/repo` pair.

**Comment by jorisvandenbossche** [comment]
@kszucs it seems that (most?) other uses of URL in our cmake files also includes a URL_HASH. Should we add that for substrait as well?

**Comment by kszucs** [comment]
I think the purpose of the `ARROW_SUBSTRAIT_REPO` variable is that the developer can quickly switch to a substrait fork with altered protobuf definitions. Introducing a checksum variable would make that less convenient.

**Comment by mbrobbel** [comment]
@jvanstraten

**Pull request closed without merging**